process/block/shardblock.go

```diff
@@ -160,12 +160,10 @@ func (sp *shardProcessor) ProcessBlock(
 		return err
 	}
 
-	numTxWithDst := sp.txCounter.getNumTxsFromPool(header.ShardID, sp.dataPool, sp.shardCoordinator.NumberOfShards())
-	go getMetricsFromHeader(header, uint64(numTxWithDst), sp.marshalizer, sp.appStatusHandler)
+	counts := sp.txCounter.getTxPoolCounts(sp.dataPool)
+	go getMetricsFromHeader(header, uint64(counts.GetTotal()), sp.marshalizer, sp.appStatusHandler)
```

Comment:
missing mutex protection for accessing the internal map. 
I would have a mutex internally in the implementation for shardedcount, so we do not care on the caller which mutex to use or not.

Reply:
Fixed.